h5p-bildetema-words-grid-view/src/components/TopicWordsGrid/TopicWordsGrid.tsx

```diff
@@ -5,14 +5,36 @@ import styles from "./TopicWordsGrid.module.scss";
 
 type TopicWordsGridProps = {
   words: WordType[];
+  showWrittenWords: boolean;
 };
 
-export const TopicWordsGrid: React.FC<TopicWordsGridProps> = ({ words }) => {
+export const TopicWordsGrid: React.FC<TopicWordsGridProps> = ({
+  words,
+  showWrittenWords,
+}) => {
+  const [textVisible, setTextVisible] = React.useState(showWrittenWords);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setTextVisible(e.target.checked);
+  };
+
+  React.useEffect(() => {
+    document
+      .getElementById("toggle")
+      ?.addEventListener("change", e => handleChange(e as any));
+
+    return () => {
+      document
+        .getElementById("toggle")
+        ?.removeEventListener("change", e => handleChange(e as any));
```

Comment:
The `removeEventListener` would remove the listener connected to a new, anonymous function reference that isn't the same as the one which listens to the `change` event. By using the `handleChange` reference, we can add and remove the same listener.

```suggestion
  const handleChange = (e: Event): void => {
    setTextVisible((e.target as HTMLInputElement).checked);
  };

  React.useEffect(() => {
    document
      .getElementById("toggle")
      ?.addEventListener("change", handleChange);

    return () => {
      document
        .getElementById("toggle")
        ?.removeEventListener("change", handleChange);
```